thingworx-import/Entities.xml

```diff
@@ -678,11 +678,162 @@ TyqZNc0vshhVwE/OK4OsFVTXATO/uu5FZ5fgfXfotCBIJBTXfJuCCwGuOahZo/N+tl7BN2PormEP
+let m2WebResultInfoTable = Resources["InfoTableFunctions"].CreateInfoTableFromDataShape({
+    infoTableName: "M2Web-Live-Data-" + ewonName + "_InfoTable",
+    dataShapeName: "M2Web-Live-Data"
+});
+
+// Build EBD command to read instant tag values via M2Web
+let m2WebHost = 'https://m2web.talk2m.com/t2mapi/get/' + ewonName + '/rcgi.bin/';
+let m2WebCommand = 'ParamForm?AST_Param=$dtIV$ftT';
+let m2WebCredentials =
+    '&t2maccount=' +
+    me.talk2MAccount +
+    '&t2musername=' +
+    me.talk2MUsername +
+    '&t2mpassword=' +
+    me.talk2MPassword +
+    '&t2mdeveloperid=' +
+    me.talk2MDeveloperID;
+let m2WebEwonCredentials =
+    '&t2mdeviceusername=' + me.ewonDeviceUsername + '&t2mdevicepassword=' + me.ewonDevicePassword;
+let m2WebCompleteURL = encodeURI(m2WebHost + m2WebCommand + m2WebCredentials + m2WebEwonCredentials);
+let params = {
+    url: m2WebCompleteURL /* STRING */ ,
+};
+
+// Perform request to M2Web and recieve response body in the following format:
+/*
+   "TagId";"TagName";"Value";"AlStatus";"AlType";"Quality"
+   2;"DEMOSTOP";0;0;0;65472
+   3;"DEMOSCENARIO";2;0;0;65472
+   4;"DEMOINUSE";1;0;0;65472
+   5;"LiftStation-A01-FLOW";8770;0;0;65472
+   6;"LiftStation-A01-PWR";1;0;0;65472
+   7;"LiftStation-B01-FLOW";9160;0;0;65472
+   8;"LiftStation-B01-PWR";1;0;0;65472
+*/
+var rawEbdResult = me.tagReadCache;
+if (useCached && me.tagReadCache.length <= 0) {
+    logger.warn("A request to read cached tag data was made, but no cached data was found! Reading via M2Web.");
+    rawEbdResult = Resources['ContentLoaderFunctions'].GetText(params);
+    me.tagReadCache = rawEbdResult;
+} else if (!useCached) {
+    rawEbdResult = Resources['ContentLoaderFunctions'].GetText(params);
+    me.tagReadCache = rawEbdResult;
+}
+
+// Split response body in to individual lines (filtering out empty entries)
+var rawEbdLines = rawEbdResult.split(/\r\n|\n|\r/).filter(function (x) { return x != ""; });
+
+// Loop through each response body line and add to info table
+for (var i = 1, rawEbdLineSegments; i < rawEbdLines.length; i++) {
+    // Split line to individual fields
+    rawEbdLineSegments = rawEbdLines[i].split(';');
+
+    // Create new info table entry with field values
+    let newInfoTableRow = {
+        TagId: parseInt(rawEbdLineSegments[0]), // INTEGER
+        TagName: rawEbdLineSegments[1], // STRING
+        Value: rawEbdLineSegments[2], // STRING
+        AlStatus: parseInt(rawEbdLineSegments[3]), // INTEGER
+        AlType: parseInt(rawEbdLineSegments[4]), // INTEGER
+        Quality: parseInt(rawEbdLineSegments[5]) // INTEGER
```

Comment:
Indexes are magic numbers.  They probably should be tagIdIndex = 0, tagNameIndex=1, etc

Reply:
I'm making this change, although I think that in a scenario like this, the line itself is self-explanatory, and separate variables for the index values are not imperative, because their meaning can be inferred by the parameter name displayed.